### ISSUE TITLE
Compilation in cygwin on windows 10

### ISSUE THREAD

**Issue by de-maas** [IanAWatson/Lilly-Medchem-Rules#10]
Hi All,

I'm trying to install this using cygwin on a windows 10 box.
After removing the -std=c++11 
and adding some packages and libraries I'm left with this compilation errors:
`
$ make
cd supporting_libraries && make install
make[1]: Entering directory '/cygdrive/d/usr/LillyMedchemRules/supporting_libraries'
cp libiwsupport.a ../lib
make[1]: Leaving directory '/cygdrive/d/usr/LillyMedchemRules/supporting_libraries'
cd Molecule && make install
make[1]: Entering directory '/cygdrive/d/usr/LillyMedchemRules/Molecule'
g++ -O2 -I . -I ../include -ffast-math -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DIW_IMPLEMENTATIONS_EXPOSED -std=c++11   -c -o mc_first_pass.o mc_first_pass.cc
In file included from smiles.h:5:0,
                 from mc_first_pass.cc:33:
../include/iwrandom.h: In member function ‘random_number_t Random_Number_Working_Storage::random_number()’:
../include/iwrandom.h:62:49: error: ‘::erand48’ has not been declared
     random_number_t random_number () { return ::erand48 ((unsigned short *) &_buffer);}
                                                 ^~~~~~~
../include/iwrandom.h:62:49: note: suggested alternative: ‘_rand48’
     random_number_t random_number () { return ::erand48 ((unsigned short *) &_buffer);}
                                                 ^~~~~~~
                                                 _rand48
make[1]: *** [<builtin>: mc_first_pass.o] Error 1
make[1]: Leaving directory '/cygdrive/d/usr/LillyMedchemRules/Molecule'
make: *** [Makefile:14: Molecule] Error 2
`

**Comment by nbehrnd** [comment]
Reading other posts and the git commit history, the problem _might_ relate to the version of `make` and `gcc`.  Using an elder Linux Xubuntu installation (18.04.4 LTS, 64 bit), for example, equipped with GNU Make 4.1 and gcc 7.5.0, I only cd'ed into the decompressed archive and run `make` without a problem.  Equally working with the test data did not cause a problem so far. Maybe the developers may help you with greater ease if the versions `make` and `gcc` accessible for you (in cygwin) are equally stated.

**Comment by IanAWatson** [comment]
Hi.

I need to fix the mess with random numbers. A long time ago, random numbers
were very difficult to get cross platform. Time has moved on and now C++
itself has perfectly find random numbers. Will be fixing this soon.

Sorry for the late reply.

It should be OK on Linux systems, but cygwin is important.

Ian

**Comment by IanAWatson** [comment]
Hi

Yes, there have been a lot of changes with gcc recently, and I do need to
keep up with that. And these days, if I want to very different versions of
Ubuntu, I should be able to do that in a Docker container.

I just looked at cygwin, and that is now a mess with the random numbers. I
will fix that asap. Random numbers are now handled just fine in the C++
language, so all the hoops we jumped through years ago are only hurting
things now.

On my own ubuntu system, I am at 19,10, but I know that a lot of people
cannot upgrade as quickly as others.

Look for some changes fairly soon - a couple of weeks realistically...

And thanks so much for your README, that was a great idea.

I really do want the software to continue to be useful, so do not hesitate
to ping me...

Thanks

Ian
